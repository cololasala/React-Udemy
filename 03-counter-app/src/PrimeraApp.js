// Functional components
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo = 'Soy un subtitulo' }) => {  // desestruccturacion para obtener saludo, subtitulo por defecto

    const holaMundo = 'Hola mundo desde component';

    return (
        <>
            <h1>{holaMundo}</h1>
            <p>Mi primer app</p>
            <p>{saludo}</p>
            <p id="sub">{ subtitulo }</p>
        </>
    )
}

PrimeraApp.propTypes = {        // permite manejar el tipo y si la property es requerida
    saludo: PropTypes.string.isRequired
}

/* PrimeraApp.defaultProps = {                  // otra manera de hacer properties por defecto
    subtitulo: "por defecto desde defaultProps"
} */

export default PrimeraApp;