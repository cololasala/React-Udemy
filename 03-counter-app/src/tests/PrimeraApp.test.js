import PrimeraApp from "../PrimeraApp";
// import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import React from 'react';


describe('Prueba <PrimeraApp/>', () => {

    /* test('debe mostrar el msj Hola soy Goku', () => {    // prueba comun sin usar la libreria 'Enzime'

        const saludo = 'Hola soy Goku';

        const { getByText } = render(<PrimeraApp saludo={saludo}/>);

        expect(getByText(saludo)).toBeInTheDocument();
    }) */

    test('debe mostrar <PrimeraApp/> correctamente', () => {
        const saludo = 'Hola soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);

        expect(wrapper).toMatchSnapshot();                  // hace que se cree un snapshot
    });

    test('debe mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola soy Goku';
        const subtitulo = 'Hola soy un subtitulo';
        const wrapper = shallow(
            <PrimeraApp 
            saludo={saludo} 
            subtitulo={subtitulo}/>
        );

        const textoParrafo = wrapper.find('#sub').text();             // esto funciona igual que el querySelector, en este caso obtengo el texto del de la etiqueta que tiene ese id
       
        expect(textoParrafo).toBe(subtitulo);
    })
})