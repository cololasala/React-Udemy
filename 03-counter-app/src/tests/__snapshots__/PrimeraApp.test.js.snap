// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <PrimeraApp/> debe mostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Hola mundo desde component
  </h1>
  <p>
    Mi primer app
  </p>
  <p>
    Hola soy Goku
  </p>
  <p
    id="sub"
  >
    Soy un subtitulo
  </p>
</Fragment>
`;
